libvast/vast/table_slice_column.hpp

```diff
@@ -34,8 +34,8 @@ class table_slice_column {
   /// Defaulted constructors, destructors, and assignment operators.
   table_slice_column() noexcept;
   ~table_slice_column() noexcept;
-  table_slice_column(const table_slice_column&) noexcept;
-  table_slice_column& operator=(const table_slice_column&) noexcept;
```

Comment:
This can stay noexcept if you move the `= default` into the header. Having it defaulted explicitly in a single TU only works with recent GCC versions.

Reply:
Isn't `noexcept` wrong because we need to allocate for the strings in the `qualified_record_field`?

Reply:
How do you want to recover from an allocation failure? We terminate anyways if we encounter a `std::bad_alloc`.

Reply:
I moved the defaulting to the header file instead to get GCC9 to compile the code.